src/test/java/org/StudentRepository/StudentRepositoryUnitTest.java

```diff
@@ -0,0 +1,129 @@
+package org.StudentRepository;
+
+import org.StudentRepository.exceptions.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StudentRepositoryUnitTest {
+
+    private StudentRepository repo;
+    private Student student1;
+    private Student student2;
+
+    @BeforeEach
+    void setUp() {
+        repo = new StudentRepository();
+        try {
+            student1 = new Student("Irina", "Popescu", LocalDate.of(2003, 4, 12), "F", "111");
+        } catch (InvalidStudentException e) {
+            throw new RuntimeException(e);
+        }
+        try {
+            student2 = new Student("Marius", "Pop", LocalDate.of(1997, 2, 26), "M", "222");
+        } catch (InvalidStudentException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    void testAddStudent_Success() {
+        try {
+            repo.addStudent(student1);
+        } catch (InvalidStudentException e) {
+            fail("Unexpected exception: " + e.getMessage());
+        }
+        try {
+            assertEquals(1, repo.getStudentsByAge("22").size());
+        } catch (ValidationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    @Test
+    void testAddStudent_DuplicateCnp_ThrowsException() throws InvalidStudentException {
```

Comment:
please use the usual java camelCase naming convention
```suggestion
    void testAddStudentDuplicateCnpThrowsException() throws InvalidStudentException {
```